boofuzz/sessions.py

```diff
@@ -1063,7 +1065,14 @@ def transmit_fuzz(self, sock, node, edge, callback_data):
     def build_webapp_thread(self, port=constants.DEFAULT_WEB_UI_PORT):
         app.session = self
         http_server = HTTPServer(WSGIContainer(app))
-        http_server.listen(port)
+        while True:
+            try:
+                http_server.listen(port)
+            except:
```

Comment:
Catch the specific exception by name to avoid silently masking other errors. E.g. `except SocktException:` (I don't know what the actual exception name is off hand)

Reply:
Something like this should work:
``` python
from socket import error as socket_error
...
 def build_webapp_thread()
...
    except socket_error as s_err:
        if s_err.errno != 98: # 98 = Address already in use.
            raise s_err # reraise if other error
        port += 1
```
This would currently fail as the db_logger can't handle any logging happenign prior to a first test case open.
To fix this change `self._current_test_case_index = None` to `self._current_test_case_index = 0` in `fuzz_logger_db.py:45`